src/miral/minimal_window_manager.cpp

```diff
@@ -276,7 +297,17 @@ auto miral::MinimalWindowManager::confirm_inherited_move(WindowInfo const& windo
 
 void miral::MinimalWindowManager::advise_new_window(miral::WindowInfo const& window_info)
 {
-    self->application_selector.advise_new_window(window_info);
+    // If the window is prevented from stealing focus, swap the old focused
+    // window (now in the back) with the new window in the front.
+    //
+    // If it's a legitimate window, it'll be focused and raised via
+    // xdg-activation.
+    auto in_background = self->prevent_focus_stealing(window_info);
+
+    self->application_selector.advise_new_window(window_info, in_background);
```

Comment:
1. It looks odd to call a function on `self` to get a value passed to a function on `self` (shouldn't all this logic be in `self`?); and,
2. `in_background` isn't the clearest name (as evidenced by the long comment explaining it) and could be `const`